tests/func/experiments/test_experiments.py

```diff
@@ -94,27 +94,33 @@ def test_update_with_pull(tmp_dir, scm, dvc, mocker):
 
 
 @pytest.mark.parametrize(
-    "change, expected",
+    "changes, expected",
     [
-        ["foo.1.baz=3", "foo: [bar: 1, baz: 3]"],
-        ["foo.0=bar", "foo: [bar, baz: 2]"],
-        ["foo.1=- baz\n- goo", "foo: [bar: 1, [baz, goo]]"],
+        [["foo=baz"], "{foo: baz, goo: {bag: 3}}"],
+        [["foo=baz,goo=bar"], "{foo: baz, goo: bar}"],
+        [["goo.bag=4"], "{foo: [bar: 1, baz: 2], goo: {bag: 4}}"],
+        [["foo[0]=bar"], "{foo: [bar, baz: 2], goo: {bag: 3}}"],
```

Comment:
Note the change in syntax, which I think is more in line with [your future plans](https://github.com/iterative/dvc/wiki/Parametrization#expression-syntax).